pkg/cmd/cli/cmd/login/logout.go

```diff
@@ -123,25 +123,43 @@ func (o LogoutOptions) RunLogout() error {
 		return err
 	}
 
+	// attempt to always delete token from local config
+	// before deleting using api.
+	tokenDeleted, err := o.deleteTokenFromConfig()
+	if err != nil {
+		return err
+	}
+
 	if err := client.OAuthAccessTokens().Delete(token); err != nil {
+		// print successful removal of token from local
+		// config before printing error from server.
+		if tokenDeleted {
+			fmt.Printf("Successfully removed user token from local config.\n\n")
+		}
 		return err
 	}
 
+	fmt.Fprintf(o.Out, "Logged %q out on %q\n", userInfo.Name, o.Config.Host)
+
+	return nil
+}
+
+func (o LogoutOptions) deleteTokenFromConfig() (bool, error) {
 	newConfig := *o.StartingKubeConfig
+	tokenDeleted := false
 
 	for key, value := range newConfig.AuthInfos {
-		if value.Token == token {
+		if value.Token == o.Config.BearerToken {
 			value.Token = ""
 			newConfig.AuthInfos[key] = value
+			tokenDeleted = true
```

Comment:
Doesn't this just mean we modified the in-memory copy of the config? Whereas the `kclientcmd.ModifyConfig` call later is what actually modifies the config on disk?